### DISCUSSION TITLE
Introductions: Fireproof user and developer community

### DISCUSSION THREAD

**Discussion by jchris** [fireproof-storage/fireproof#4] [Announcements]
## 🔥 Welcome!
  As a Fireproof user, you'll find other people here who have strong opinions about things like JSON, React, databases, SQL, CRDTs, JavaScript, TypeScript, and IPFS.

Please join us by:
 * [Hop into the Discord](https://discord.gg/cCryrNHePH)
  * Sharing your projects that use Fireproof (this makes a huge difference!)
  * Asking for help, about the roadmap, or suggesting documentation ideas.
  * "Wouldn't it be cool if..." is especially welcomed.
  * Creating a welcoming environment for everyone 💪.

  To get started, please feel free to comment below introducing yourself and how you use Fireproof.

**Comment by jchris** [comment]
I'm Chris, I started dreaming of Fireproof in about 2013, and working on it in February of 2023, so that's a decade dreaming, and six months working on it. Most of that time was on the first version, [Fireproof alpha](https://github.com/fireproof-storage/fireproof-alpha), which was written in the spirit of Fred Brook's "write one to throw away" as an opportunity to learn the relevant algorithms and how it felt to use in React apps. With the discovery of the [useLiveQuery React hook API](https://use-fireproof.com/docs/react-hooks/use-live-query), I switched focus to the integrity of the storage engine and rebuilt Fireproof beta (starting with 0.11.0 in this repo) in TypeScript with code quality in mind.

Before this, I was one of the creators of Couchbase Mobile, a peer-to-peer database used for workloads like [United Airlines' crew scheduling](https://www.couchbase.com/customers/united-airlines/) and [GE Digital](https://www.couchbase.com/customers/ge/). Fireproof brings this flexibility and reliability to the web, with a focus on easy adoption (developer experience, deployment requirements).

When I'm not debugging CRDTs I keep bonsai trees, [make wild music](https://soundcloud.com/jchris), and ride my cargo ebike.

**Comment by dhruvsoni1802** [comment]
Hey everyone I am Dhruv, a 4th year engineering student from India with a passion creating full stack applications and smart contracts and ofc a love for pizzas! I am an intern at fireproof and currently building small cool applications and learning more and more about the codebase and its technical specifications. 

I was previously an intern at Motorola Solutions. Got into fireproof because being a react developer i found its use-cases pretty handy and its developer experience and onboarding is smooth. Chris is  the best mentor I have got and he has been really helpful in this journey. 

Will continue to work on building more and more applications that integrates fireproof for the next few months to understand more about its usecases. Ping me or chris up if you have a super cool idea about integrating fireproof with your applications:)

**Comment by DougAnderson444** [comment]
Hi I'm Doug 👋

I've been looking for robust ways for people to share data with each other since 2018 when I moved for the millionth time due to being in the Navy and had to update 99 accounts as to my new whereabouts and contact info 🤦‍♂️.  I've also been into software and the internet since the 1990s, JavaScript since the jQuery early days!

I actually looked at [mikeal's car-transactions](https://github.com/mikeal/car-transaction) years ago even [created a prototype](https://github.com/DougAnderson444/ipld-car-txs) to build DAGs on top of, so I'm happy jchris built fireproof so I didn't have to keep banging my head against the wall.

I also love Rust and have contributed to `rust-libp2p` specifically the `webRTC` feature for the browsers. "Wouldn't it be cool if" fireproof could connect to a home `libp2p` node over `WebRTC` to backup your browser data in the same way `web3.storage` does?

I'm also looking for ways to [share read-only](https://github.com/fireproof-storage/fireproof/issues/105) versions of databases, as we don't always want to hand out write access.

I've [written a ton of other cool stuff](https://github.com/DougAnderson444) in JS, Rust, and WebAssembly a lot of which uses cryptography in some shape. I've built crypto wallets, encryption libraries, and I'm interested in the neat things we could do with fireproof keys and data sharing too.

Ultimately I'd like to point [PeerPiper.io](https://peerpiper.io/) to all my fireproof database names to build out [Drawn Systems](https://drawn.systems) so people can share visualizations of their data, notes, and other stuff in an independent and decentralized way.

**Comment by 2color** [comment]
Hey there,

I'm Daniel 👋
I'm a software developer and currently also a developer advocate for IPFS and have been active in the IPFS ecosystem for almost two years. I'm very excited about all the efforts @jchris has put into Fireproof! 

I sense that Fireproof is part of the broader local-first movement to bring agency back into the hands of users in a tractable way. It's great seeing both new and known faces (Hey @DougAnderson444). Hoping to get more time to dive deeper into Fireproof in the near future.

Keep the great work going!

**Reply by DougAnderson444** [reply]
Hey Daniel, great to see you here too!
